core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java

```diff
@@ -0,0 +1,55 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JavaGrepLambdaImp extends JavaGrepImp {
+
+  @Override
+  public List<String> readLines(File inputFile) throws IllegalArgumentException {
+    Path path = inputFile.toPath();
+    List<String> lines = new ArrayList<>();
+    try {
+      Files.lines(path).forEach(line -> lines.add(line));
+    } catch (Exception e) {
+      logger.error("Unable to read file", new IOException());
+    }
+    return lines;
+  }
+
+  @Override
+  public List<File> listFiles(String rootDir) {
+    Path path = Paths.get(rootDir);
+    List<File> files = new ArrayList<>();
+    try {
+      Files.walk(path)
+          .filter(Files::isRegularFile)
+          .forEach(file -> files.add(file.toFile()));
+    } catch (Exception e) {
+      logger.error("Unable to read directory", new IOException());
```

Comment:
you shouldn't throw a new IOExcpetion for the second argument
logger.error()
throw new RuntimeException....

so the program will stop if it unable to list files